### PR TITLE
Implement the effect handler for the select country screen

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://www.pivotaltracker.com/story/show/168980808

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryEffectHandlerTest.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class SelectCountryEffectHandlerTest {
+
+  private val repository = mock<AppConfigRepository>()
+
+  private val effectHandler = SelectCountryEffectHandler.create(repository, TrampolineSchedulersProvider())
+  private val testCase = EffectHandlerTestCase(effectHandler)
+
+  @Before
+  fun setUp() {
+    whenever(repository.fetchAppManifest()) doReturn Single.never<ManifestFetchResult>()
```
Comment (ragunathjawahar) [inline app/src/test/java/org/simple/clinic/selectcountry/SelectCountryEffectHandlerTest.kt]: Is this necessary? Is it not okay to just stub out calls that are being exercised in the appropriate test case?

File: app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt [hunk 2]
```diff
@@ -1,10 +1,11 @@
 package org.simple.clinic.selectcountry
 
 import org.simple.clinic.appconfig.Country
+import org.simple.clinic.util.ResolvedError
 
 data class SelectCountryModel(
     val countries: List<Country>?,
-    val manifestFetchError: ManifestFetchError?,
+    val manifestFetchError: ResolvedError?,
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt]: Can we model these as domain errors instead? (Details discussed offline).

File: app/src/main/java/org/simple/clinic/appconfig/ManifestFetchResult.kt [hunk 3]
```diff
@@ -1,5 +1,9 @@
 package org.simple.clinic.appconfig
 
+import org.simple.clinic.util.ResolvedError
+
 sealed class ManifestFetchResult
 
 data class ManifestFetchSucceeded(val countries: List<Country>) : ManifestFetchResult()
+
+data class ManifestFetchError(val error: ResolvedError): ManifestFetchResult()
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/appconfig/ManifestFetchResult.kt]: Same here. Please use domain errors.

File: app/src/test/java/org/simple/clinic/selectcountry/SelectCountryEffectHandlerTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/selectcountry/SelectCountryEffectHandlerTest.kt]: Done.

File: app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/selectcountry/SelectCountryModel.kt]: Done.

File: app/src/main/java/org/simple/clinic/appconfig/ManifestFetchResult.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/appconfig/ManifestFetchResult.kt]: Done.

**Pull request merged**